fix: fixed functioning of contact us button in footer

MahendraDani opened the pull request:
<!--- THESE ARE COMMENTS, AND WON'T BE VISIBLE, DON'T WORRY 

🔴🔴🔴 PLEASE USE PROPER PR TITLE, IT'S SUPER IMPORTANT 
🔴🔴🔴 ALL LOWER CASE CHARACTERS ONLY 

EXAMPLES👇🏻👇🏻  

feat: added new footer links
fix: changes to the buggy buttons
docs: upgrades to the readme file
chore(deps): bumping up the dependencies
chore(refactor): refactored legacy codes for server

🔴🔴🔴 MAKE SURE YOU FOLLOW THESE !  -->


<!--- ADD YOUR ISSUE NUMBER LIKE #11 (NO SPACES BETWEEN # & ISSUE NUMBER) -->

closes #1131 
closes #1141


### 👷🏻 Changes made  
<!--- A clear and concise (minimum 2 line) description of what you have done to successfully close your assigned issue.--->
1. Fixed the modal component. Modal component can now have dynamic children as per the need.
2. On clicking the `contact us` button , a modal opens where user can report a problem. The Endpoint is `/user/report`.


### 📸 Screenshots 
<!--- (Remove this section if no SS) --> 

https://github.com/MilanCommunity/Milan/assets/116940083/50bbb5e4-e922-4b57-9205-7ccee6eaf357

The above recording shows three things-
1. Proper functioning of `contact us` button and `modal` both.
2. Sucessfull API call to **report problem route.**
3. Proper functioning of database, where the report is stored as expected.

tamalCodes commented on src/components/Modal/Modal.jsx:
We are using two different fonts `Inter` for the headings and `Poppins` for the normal text and buttons.
Please make sure that the fonts are taken care of along with the color of the text in the modal.

MahendraDani replied:
Would you please tell me the text color hex codes ?

tamalCodes replied:
For the heading, it would be `#28183b`, and for the normal text we have it as `#000000`

MahendraDani replied:
Thanks, will make the necessary changes

MahendraDani replied:
@tamalCodes Made changes and pushed in https://github.com/MilanCommunity/Milan/pull/1140/commits/e118422d6932bc8a399794f118937dddf7e921c4

MahendraDani commented:
## Applied changes
1. Added font-family *Inter* to headers and *Poppins* to normal text in modals.
2. Removed irrelevant text from console, fixes #1141 

## Before
![Screenshot (25)](https://github.com/MilanCommunity/Milan/assets/116940083/190e6ef5-479a-4be9-9aaf-079837141c38)

## After
![Screenshot (24)](https://github.com/MilanCommunity/Milan/assets/116940083/0606f0f6-24af-4e41-93c2-b1fb40b36059)

@tamalCodes If there are any other changes please let me know.

tamalCodes commented on src/components/Footer/Footer.css:
```diff
@@ -24,18 +24,20 @@
 .reportModalHeader > h1 {
-  font-family: Montserrat, sans-serif;
-  color: #4e4e4e;
+  font-family: "Inter";
```
```suggestion
  font-family: "Inter",sans-serif;
```

tamalCodes commented on src/components/Footer/Footer.css:
```diff
@@ -24,18 +24,20 @@
 
 .reportModal > h2 {
-  color: #4e4e4e;
+  font-family: "Poppins";
```
```suggestion
  font-family: "Poppins"sans-serif;
```

tamalCodes commented on src/components/Footer/Footer.css:
```diff
@@ -24,18 +24,20 @@
 }
 
 #reportForm {
+  font-family: "Poppins";
```
```suggestion
  font-family: "Poppins"sans-serif;
```

tamalCodes reviewed (approved):
# Amazing work 🎉
**The changes look good to me and will be merged soon.**

Do follow [Tamal](https://github.com/tamalCodes) for more Opensource fun projects and don't forget to drop a star so that you get updated about our latest [releases](https://github.com/MilanCommunity/Milan/releases) (we will tag you and mention your work) and also a [shoutout](https://www.linkedin.com/feed/update/urn:li:activity:7073574804519354368/) on social media (LinkedIn and Twitter) !

Happy Opensource 🚀.

MahendraDani commented:
@tamalCodes Thanks for merging this pull request.
If any issues need addressing I am up for grabs!
